reactive/docker.py

```diff
@@ -187,6 +224,46 @@ def install_from_upstream_apt():
     apt_install(['docker-engine'], fatal=True)
 
 
+def install_from_nvidia_apt():
+    ''' Install cuda docker from the nvidia apt repository. This is a pyton adaptation of
+    the shell script found at https://get.docker.com/ '''
+    status_set('maintenance', 'Installing docker-engine from Nvidia PPA.')
+
+    ## Get the server and key in the apt-key management tool.
```

Comment:
Could we try to reuse code either in the apt-layer or in the charm helpers for this repo management? The logstash charm has a config variable for the repo https://github.com/juju-solutions/layer-logstash/blob/85d85715b38f81b2e72dea30b5463aaf2284c5ef/config.yaml  and uses charmhelpers.fetch package to load the repo https://github.com/juju-solutions/layer-logstash/blob/85d85715b38f81b2e72dea30b5463aaf2284c5ef/reactive/logstash.py#L27 https://pythonhosted.org/charmhelpers/api/charmhelpers.fetch.html#charmhelpers.fetch.configure_sources . There is also the apt layers add_source https://git.launchpad.net/layer-apt/tree/lib/charms/apt.py#n37 would it make sense to have the right repos for each arch we support. The drawback is that there will be a lot of config parameters. The good thing is that we would have these configurable so we will have a proper way to address issues like: https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/452 . What do others think about this?